nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -143,18 +165,27 @@ public Batch getBatch(String batchId) {
 
         // create the batch
         Batch batch = new Batch(batchId, parameters, getName(), getTransientStore());
-        AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest().withRoleSessionName(batch.getKey())
-                                                                     .withRoleArn(roleArn);
-        Credentials credentials = stsClient.assumeRole(assumeRoleRequest).getCredentials();
+
+        String bucketPath = Paths.get(bucketPrefix, policyTemplateMatcher.replaceAll(batchId)).toString();
+
+        GetFederationTokenRequest federationTokenRequest = new GetFederationTokenRequest()
+                .withPolicy(bucketPath)
+                .withName(
+                        MessageFormat.format("{0}_{1}", batchId, ClientLoginModule.getCurrentPrincipal()
```

Comment:
Why not using `String#format` ? Also format is not correct, `;` is on a new line.

Reply:
It's a personal preference. I think it's more human-readable this way.

I'll fix the `;`

Reply:
`MessageFormat.format` if overkill when you can do just `batchId + '_' + originatingUser` which is also very readable. Please simplify.